sys/kern/gmon.c

```diff
@@ -0,0 +1,59 @@
+#include <sys/param.h>
+#include <sys/interrupt.h>
+#include <sys/klog.h>
+#include <sys/kmem.h>
+#include <sys/gmon.h>
+#include <sys/param.h>
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/mimiker.h>
+#include <machine/vm_param.h>
+#include <sys/types.h>
+
+gmonparam_t _gmonparam = {.state = GMON_PROF_NOT_INIT};
+gmonhdr_t _gmonhdr = {.profrate = CLK_TCK};
+/* The macros description are provided in gmon.h */
+void init_kgprof(void) {
+#if KGPROF == 0
+  return;
+#endif
```

Comment:
Is it possible to move that conditional compilation into header file like we do for KASAN?

```C
#if KGPROF
void init_kgprof(void);
#else
#define init_kgprof() __nothing
#endif
```